build_projects/dotnet-host-build/PublishTargets.cs

```diff
@@ -83,21 +85,81 @@ public static BuildTargetResult FinalizeBuild(BuildTargetContext c)
             {
                 string targetContainer = $"{Channel}/Binaries/Latest/";
                 string targetVersionFile = $"{targetContainer}{CommitHash}";
+                string flagFinalizationInProgress = $"{targetContainer}/Finalizing";
                 string semaphoreBlob = $"{Channel}/Binaries/sharedFxPublishSemaphore";
+                
                 AzurePublisherTool.CreateBlobIfNotExists(semaphoreBlob);
 
-                string leaseId = AzurePublisherTool.AcquireLeaseOnBlob(semaphoreBlob);
```

Comment:
It looks like we already have the looping logic you added below in `AcquireLeaseOnBlob` by passing the `maxWaitDefault` parameter. Can't we just pass `TimeSpan.FromSeconds(1800)` into this function call to extend the timeout?